SwrveCommonSDK/src/main/java/com/swrve/sdk/SwrveImp.java

```diff
@@ -933,33 +933,38 @@ protected void unbindAndShutdown() {
         }
     }
 
-    protected void removeCurrentDialog() {
+    protected void removeCurrentDialog(Activity callerActivity) {
```

Comment:
Yeah, but the first activity is not the owner of the dialog. The dialog should only be removed if the owner activity is finishing. dialog.getOwnerActivity().isFinishing().